### PR TITLE
Add the ImageJ Macro Language

### PR DESCRIPTION
**Pull request by davidbrown2324**
I am adding the ImageJ Macro language widely used by scientists to handle images, particularly microscopy data.
The macro language is saved in .ijm files. It would be great to see it recognised by GitHub.
I chose color: "#99AAFF# from the ImageJ Macro web documentation. It did not appear elsewhere in the list.

Checklist:
- [x]  I am adding a new language.
- [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
Search results for extension:ijm = [7K](
https://github.com/search?q=extension%3Aijm&type=Code)
- [x] I have included a real-world usage sample for all extensions added in this PR:

Sample source(s):
1. [simple.ijm](https://imagej.nih.gov/ij/macros/examples/RoiFunctionsDemo.txt)
1. [batch of ratiometric FRET using IO settings.ijm](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/blob/master/virginie/macros/batch%20of%20ratiometric%20FRET%20using%20IO%20settings.ijm) 

Sample license(s):
 1. [ImageJ License](https://imagej.net/disclaimer.html)
 1. [MIT](https://github.com/MontpellierRessourcesImagerie/imagej_macros_and_scripts/blob/master/LICENSE)

- [ ] I have included a syntax highlighting grammar
- [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2381,6 +2381,15 @@ Ignore List:
   codemirror_mode: shell
   codemirror_mime_type: text/x-sh
   language_id: 74444240
+ImageJ Macro:
+  type: programing
```
Comment (Nixinova) [inline lib/linguist/languages.yml:2385]: Typo there, causing a failed test: `invalid type: programing (ArgumentError)`. Should be "programming" with 2 Ms.

Comment (davidbrown2324) [reply lib/linguist/languages.yml]: Thanks Nixinova. I am one step closer.

**Comment by Nixinova** [comment]
Samples need to be in a folder with the exact same name as the language, ImageJ Macro. Logs currently throw an error saying `Expected: "ImageJ"  Actual: "ImageJ Macro"`

**Comment by Nixinova** [comment]
The other error is a missing language ID: use https://gist.github.com/Nixinova/6817e82fffe85f0b33491d32ad8742eb to generate it, replacing "lang" with "ImageJ Macro".

**Comment by davidbrown2324** [comment]
At this point I am quite lost. 

Could you tell me what are the three Ruby checks for? 

I am very sorry for all the pings. I have spent weeks trying to work out how to get this right on my own machine, but as soon as I submitted the pull request it feels like a totally different approach.

Is it running ruby script stored within the repo? I am having problems installing all dependencies on my machine. If not, then I have no idea.

**Comment by Nixinova** [comment]
Green tick, gj :+1:

**Comment by lildude** [comment]
Can you please add a link to the source and state the license of the `simple.ijm` sample you've included to the OP.

**Comment by davidbrown2324** [comment]
> Can you please add a link to the source and state the license of the `simple.ijm` sample you've included to the OP.

I have updated the original comment now, with a link and a source (does it matter that the source is a text file I converted to the .ijm format?).

Thanks for your help an patience.

**Comment by lildude** [comment]
> (does it matter that the source is a text file I converted to the .ijm format?)

Nope.

Thanks for updating the OP.

**Review by lildude — APPROVED** [review]

**Comment by imagejan** [comment]
Thanks for this, the ImageJ Macro language is nicely detected in the language bar statistics. However, when I click the linked search for ImageJ Macro language, the results are empty, see e.g. https://github.com/cleterrier/Utils/search?l=imagej-macro

Is that search functionality also located in this repository, or are changes required somewhere else to make the search work?

**Comment by lildude** [comment]
> Is that search functionality also located in this repository, or are changes required somewhere else to make the search work?

Search uses [go-enry](https://github.com/go-enry/go-enry) for the language detection which feeds in Linguist releases but these tend to lag a few months. Case in point, https://github.com/go-enry/go-enry/pull/52 is still open to update to v7.14.0 however we've already released v7.15.0, so it should probably be closed and a newer PR opened.

I don't summit a PR to go-enry when I make a new Linguist release but I'm tempted to start doing so to head off questions like yours.

**Pull request merged**